quicfire_tools/inputs.py

```diff
@@ -722,3 +745,356 @@ def from_file(cls, directory: str | Path):
+        FuelParams class with named fuel parameters for a uniform grid.
+        Defaults to None unless fuel_flag = 1
+    ignition_type: IgnitionType
+        Ignitions shape or source. @anthony
+    ignitions_per_cell: int
+        Number of ignition per cell of the fire model. Recommended max value
+        of 100
+    firebrand_flag : int
+        Firebrand flag, 0 = off; 1 = on. Recommended value = 0 ; firebrands
+        are untested for small scale problems
+    auto_kill : int
+        Kill if the fire is out and there are no more ignitions or firebrands
+        (0 = no, 1 = yes)
+    eng_to_atm_out : int
+        Output flag [0, 1]: gridded energy-to-atmosphere
+        (3D fire grid + extra layers)
+    react_rate_out : int
+        Output flag [0, 1]: compressed array reaction rate (fire grid)
+    fuel_dens_out : int
+        Output flag [0, 1]: compressed array fuel density (fire grid)
+    QF_wind_out : int
+        Output flag [0, 1]: gridded wind (u,v,w,sigma) (3D fire grid)
+    QU_wind_inst_out : int
+        Output flag [0, 1]: gridded QU winds with fire effects, instantaneous
+        (QUIC-URB grid)
+    QU_wind_avg_out : int
+        Output flag [0, 1]: gridded QU winds with fire effects, averaged
+        (QUIC-URB grid)
+    fuel_moist_out : int
+        Output flag [0, 1]: compressed array fuel moisture (fire grid)
+    mass_burnt_out : int
+        Output flag [0, 1]: vertically-integrated % mass burnt (fire grid)
+    firebrand_out : int
+        Output flag [0, 1]: firebrand trajectories. Must be 0 when firebrand
+        flag is 0
+    emissions_out : int
+        Output flag [0, 5]: compressed array emissions (fire grid):
+            0 = do not output any emission related variables
+            1 = output emissions files and simulate CO in QUIC-SMOKE
+            2 = output emissions files and simulate PM2.5 in QUIC- SMOKE
+            3 = output emissions files and simulate both CO and PM2.5 in
+                QUIC-SMOKE
+            4 = output emissions files but use library approach in QUIC-SMOKE
+            5 = output emissions files and simulate both water in QUIC-SMOKE
+    radiation_out : int
+        Output flag [0, 1]: gridded thermal radiation (fire grid)
+    intensity_out : int
+        Output flag [0, 1]: surface fire intensity at every fire time step
+    """
+    filename: str = Field("QUIC_fire.inp", allow_mutation=False)
+    nz: PositiveInt
+    time_now: PositiveInt
+    sim_time: PositiveInt
+    output_times: PositiveInt | OutputTimes
+    ignition_type: IgnitionType = IgnitionType(ignition_flag=6)
+    fire_flag: Literal[0, 1] = 1
+    random_seed: PositiveInt = random.randrange(1000)
+    fire_time_step: PositiveInt = 1
+    quic_time_step: PositiveInt = 1
+    stretch_grid_flag: Literal[0, 1] = 0
+    dz: PositiveInt = 1
+    dz_array: list[PositiveFloat] = []
+    file_path: str = '""'
+    fuel_flag: Literal[1, 2, 3, 4] = 3
+    fuel_params: FuelParams | None = None
```

Comment:
See above comments about splitting these up into individual parameters. They can default to 0 if needed.

Reply:
Still not sure what you're referring to. FuelParams now?